File: programmers/hash_3.java
//위장
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String,Integer> map = new HashMap<>();
        for(int i = 0;i<clothes.length;i++){
            //옷 종류가 key, value 가 해당 옷 종류의 갯수
            if(map.get(clothes[i][1]) == null){
                map.put(clothes[i][1], 1);
            }
            else{
                map.put(clothes[i][1],map.get(clothes[i][1])+1);
            }
            
        }
        Set<String> keySet = map.keySet();
        for(String key : keySet){
            answer *= map.get(key)+1;
        }
        return answer-1;
    }
}

File: programmers/hash_3.py
#위장
def solution(clothes):
    answer = 1
    dic = {}
    for i in range(len(clothes)):
        if(clothes[i][1] in dic):
            tmp = dic.get(clothes[i][1])
            tmp += 1
            dic[clothes[i][1]] = tmp
        else:
            dic[clothes[i][1]] = 1
    dVal = dic.values()
    
    for i in dVal:
        answer *= i+1
        
    return answer-1
